libpod/container_api.go

```diff
@@ -515,6 +516,20 @@ func (c *Container) WaitForExit(ctx context.Context, pollInterval time.Duration)
 	var conmonTimer time.Timer
 	conmonTimerSet := false
 
+	conmonPidFd := -1
+	conmonPidFdTriggered := false
+
+	if c.state.ConmonPID != 0 {
+		// Track lifetime of conmon precisely using pidfd_open + poll.
+		// There are many cases for this to fail, for instance conmon is dead
+		// or pidfd_open is not supported (pre linux 5.3), so fall back to the
+		// traditional loop with poll + sleep
+		if fd, err := unix.PidfdOpen(c.state.ConmonPID, 0); err == nil {
+			conmonPidFd = fd
+			defer unix.Close(conmonPidFd)
+		}
```

Comment:
Coud logrus.Debugf(err) on failure.

Reply:
Perhaps ignore ENOSYS.

Reply:
The two "expected" errors are ENOSYS (old kernel) and EINVAL (conmon died). Maybe we should log other errors.

Reply:
Ah, no:        ESRCH  The process specified by pid does not exist.

Reply:
Added the debugf